test/common/validation.go

```diff
@@ -92,14 +94,36 @@ func (client *Client) FindAnyLogContents(podName string, contents []string) (boo
 	if err != nil {
 		return false, err
 	}
+	eventContentsSet, err := parseEventContentsFromPodLogs(string(logs))
+	if err != nil {
+		return false, err
+	}
 	for _, content := range contents {
-		if strings.Contains(string(logs), content) {
+		_, ok := eventContentsSet[content]
+		if ok {
 			return true, nil
 		}
 	}
 	return false, nil
 }
 
+func parseEventContentsFromPodLogs(logs string) (map[string]bool, error) {
```

Comment:
maybe add a comment to this func so that is easy to understand what it returns and what it does?

Reply:
Comments on functions or variables should start with the function/variable's name. E.g.

```golang
// parseEventContentsFromPodLogs extracts the contents of an event from a Pod's logs. 
// Example log entry: ...
```